rpc/block_svc_routines.c

```diff
@@ -4491,7 +4502,16 @@ block_create_v2_1_svc_st(blockCreate2 *blk, struct svc_req *rqstp)
 
   convertTypeCreate2ToCreate(blk, &blk_v1);
 
-  return block_create_common(&blk_v1, rbsize, blk->prio_path);
+  if (len > 0 && len <= HOST_NAME_MAX) {
+    GB_ALLOC_N(volServer, HOST_NAME_MAX);
+    strncpy(volServer, blk->xdata.xdata_val, len);
+    if (!strcmp(volServer, "localhost")) {
```

Comment:
Can this move before allocation, with just blk->xdata.xdata_val? Saves a dummy alloc and free.

Reply:
Done